### DISCUSSION TITLE
Using regexp rules together with google java format

### DISCUSSION THREAD

**Discussion by sijonelis** [diffplug/spotless#1781] [Q&A]
I am trying to set up my own regexp rules (to capture empty lines if that is important), however, i seem to be hitting a configuration issue or conflict with google java format. 

If I run either googleJavaFormat or regexp, both seem to be working, but if i try to run both as shown below, I hit a googlejavaformat error:
```
[ERROR] Step 'google-java-format' found problem in 'src\test\java\com\foo\bar\SomeClass.java':
1:2: error: class, interface, enum, or record expected
com.google.googlejavaformat.java.FormatterException: 1:2: error: class, interface, enum, or record expected
    at com.google.googlejavaformat.java.FormatterException.fromJavacDiagnostics (FormatterException.java:51)
    at com.google.googlejavaformat.java.Formatter.format (Formatter.java:149)
```
Do you have any idea what i might be doing wrong? 

The relevant part of my spotless config can be found below:
```
<formats>
	<format>
		<includes>
			<include>*.md</include>
			<include>.gitignore</include>
		</includes>
		<trimTrailingWhitespace/>
		<endWithNewline/>
		<indent>
			<tabs>true</tabs>
			<spacesPerTab>4</spacesPerTab>
		</indent>
	</format>
</formats>
<java>
	<includes>
		<include>src/main/java/**/*.java</include>
		<include>src/test/java/**/*.java</include>
		<include>src/integrationTest/java/**/*.java</include>
	</includes>
	<formatAnnotations/>
	<googleJavaFormat>
		<version>${google-java-format.version}</version>
		<reflowLongStrings>true</reflowLongStrings>
	</googleJavaFormat>
	<replaceRegex>
		<name>Replace a's with b's</name>
		<searchRegex>(a)</searchRegex>
		<replacement>b</replacement>
	</replaceRegex>
</java>```

**Comment by Marcono1234** [comment]
> If I run either googleJavaFormat or regexp, both seem to be working

What do you consider 'working' for your `replaceRegex` config? It appears it replaces any 'a' with 'b', so for example
```java
package test;

class MyClass {
}
```

becomes (corrupted):
```java
pbckbge test;

clbss MyClbss {
}
```

If I understand it correctly Google Java Format requires that your code is syntactically correct. But it appears Spotless runs the `replaceRegex` step first (or runs the steps multiple times?), and after that the source code might not be valid anymore.

Maybe your real configuration is also by accident corrupting the Java source code?
